tests/tests_pytorch/trainer/properties/test_estimated_stepping_batches.py

```diff
@@ -135,16 +136,29 @@ def test_num_stepping_batches_gpu(trainer_kwargs, estimated_steps, monkeypatch):
     assert trainer.estimated_stepping_batches == estimated_steps
 
 
+@RunIf(tpu=True, standalone=True)
+def test_num_stepping_batches_with_tpu_single():
+    """Test stepping batches with the single-core TPU strategy."""
+    trainer = Trainer(accelerator="tpu", devices=1, max_epochs=1)
+    model = BoringModel()
+    trainer._data_connector.attach_data(model)
+    trainer.strategy.connect(model)
+    assert trainer.estimated_stepping_batches == len(model.train_dataloader())
+
+
 @RunIf(tpu=True)
-@pl_multi_process_test
-@pytest.mark.parametrize("devices,estimated_steps", [([1], 64), (8, 8)])
-def test_num_stepping_batches_with_tpu(devices, estimated_steps):
-    """Test stepping batches with TPU training which acts like DDP."""
-    trainer = Trainer(accelerator="tpu", devices=devices, max_epochs=1)
+@mock.patch(
+    "pytorch_lightning.strategies.tpu_spawn.TPUSpawnStrategy.root_device",
+    new_callable=PropertyMock,
+    return_value=torch.device("xla:0"),
+)
```

Comment:
Reviewer: Since we never iterate over the dataloader, it is ok to mock this here. 
This is needed because the `estimated_stepping_batches` will call the `reset_dataloader()` method and then in turn call the `TPUSpawnStrategy.process_dataloader`, which wraps the dataloader and requires the device.